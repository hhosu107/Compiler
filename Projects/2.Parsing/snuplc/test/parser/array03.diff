24,25c24,25
<         := <INVALID>
<           [ @A        <array 5  of <int>> ] <INVALID>
---
>         := <int>
>           [ @A        <array 5  of <int>> ] <int>
30c30
<           [ @A        <array 5  of <int>> ] <INVALID>
---
>           [ @A        <array 5  of <int>> ] <int>
33c33,34
<           [ @A        <array 5  of <int>> ] <array 5  of <int>>
---
>           &() <ptr(4) to <array 5  of <int>>>
>             [ @A        <array 5  of <int>> ] <array 5  of <int>>
47,48c48,49
<             := <INVALID>
<               [ %P        <ptr(4) to <array  of <int>>> ] <INVALID>
---
>             := <int>
>               [ %P        <ptr(4) to <array  of <int>>> ] <int>
64c65
<               < <int>
---
>               < <bool>
69c70
<                 [ @A        <array 5  of <int>> ] <INVALID>
---
>                 [ @A        <array 5  of <int>> ] <int>
