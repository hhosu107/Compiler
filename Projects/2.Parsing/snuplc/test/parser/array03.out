parsing 'array03.mod'...
successfully parsed.
  AST:
    CAstScope: 'array03'
      symbol table:
        [[
          [ @A        <array 5  of <int>>           ]
          [ *DIM(<ptr(4) to <NULL>>,<int>) --> <int>           ]
          [ *DOFS(<ptr(4) to <NULL>>) --> <int>           ]
          [ *ReadInt() --> <int>           ]
          [ *Set(<ptr(4) to <array  of <int>>>,<int>) --> <NULL>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<int>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteStr(<ptr(4) to <array  of <char>>>) --> <NULL>           ]
          [ @i        <int>           ]
          [ *print() --> <NULL>           ]
        ]]
      statement list:
        call [ *WriteInt(<int>) --> <NULL> ] <NULL>
          11111111 <int>
        call [ *WriteLn() --> <NULL> ] <NULL>
        call [ *print() --> <NULL> ] <NULL>
        := <INVALID>
          [ @A        <array 5  of <int>> ] <INVALID>
            2 <int>
          3 <int>
        := <int>
          [ @i        <int> ] <int>
          [ @A        <array 5  of <int>> ] <INVALID>
            2 <int>
        call [ *Set(<ptr(4) to <array  of <int>>>,<int>) --> <NULL> ] <NULL>
          [ @A        <array 5  of <int>> ] <array 5  of <int>>
          [ @i        <int> ] <int>
        call [ *WriteInt(<int>) --> <NULL> ] <NULL>
          22222222 <int>
        call [ *WriteLn() --> <NULL> ] <NULL>
        call [ *print() --> <NULL> ] <NULL>
      nested scopes:
        CAstScope: 'Set'
          symbol table:
            [[
              [ %P        <ptr(4) to <array  of <int>>>               ]
              [ %v        <int>               ]
            ]]
          statement list:
            := <INVALID>
              [ %P        <ptr(4) to <array  of <int>>> ] <INVALID>
                [ %v        <int> ] <int>
              1 <int>
          nested scopes:
            empty.
        
        CAstScope: 'print'
          symbol table:
            [[
              [ $i        <int>               ]
            ]]
          statement list:
            := <int>
              [ $i        <int> ] <int>
              0 <int>
            while cond
              < <int>
                [ $i        <int> ] <int>
                5 <int>
            while-body
              call [ *WriteInt(<int>) --> <NULL> ] <NULL>
                [ @A        <array 5  of <int>> ] <INVALID>
                  [ $i        <int> ] <int>
              call [ *WriteLn() --> <NULL> ] <NULL>
              := <int>
                [ $i        <int> ] <int>
                add <int>
                  [ $i        <int> ] <int>
                  1 <int>
          nested scopes:
            empty.
        
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -oarray03.mod.ast.pdf array03.mod.ast.dot


Done.
